app/validators/sections/section_validator.py

```diff
@@ -96,6 +96,15 @@ def validate_blocks(self, group_id):
 
             self.validate_question(block)
             self.validate_variants(block)
+            # Repeating blocks must be validated here instead of from ListCollectorValidator as the latter cannot do standard block validation
+            for repeating_block in block.get("repeating_blocks", []):
+                block_validator = get_block_validator(
+                    repeating_block, self.questionnaire_schema
+                )
+                self.errors += block_validator.validate()
+
+                self.validate_question(repeating_block)
+                self.validate_variants(repeating_block)
```

Comment:
I think it would be cleaner to read and follow if you do `self.validate_repeating_blocks(block)`.

Reply:
Made a function which is explicitly only called for list collector block validators

Reply:
I think it is better if we do this on the method level, so it is same  everywhere, same as we do in `self.validate_question()`:
So:
```python
question = block_or_variant.get("question")
        if question:
```
Something like:
```python
repeating_blocks = block.get("repeating_blocks", []):
     if repeating_blocks:
```
Thoughts?

Reply:
Yep agree that would be clean and consistent, have updated to reflect this. I have also removed the check for the block validator being a ListCollectorValidator as it seemed redundant given the following method is checking for the repeating blocks and only validating if they exist - only list collectors would have them.

Reply:
I like this change 👍